packages/cloud_firestore/cloud_firestore/lib/src/document_snapshot.dart

```diff
@@ -41,13 +41,13 @@ class DocumentSnapshot {
   /// Data can be accessed by providing a dot-notated path or [FieldPath]
   /// which recursively finds the specified data. If no data could be found
   /// at the specified path, a [StateError] will be thrown.
-  dynamic get(dynamic field) =>
+  dynamic get(Object field) =>
       _CodecUtility.valueDecode(_delegate.get(field), _firestore);
 
   /// Gets a nested field by [String] or [FieldPath] from this [DocumentSnapshot].
   ///
   /// Data can be accessed by providing a dot-notated path or [FieldPath]
   /// which recursively finds the specified data. If no data could be found
   /// at the specified path, a [StateError] will be thrown.
-  dynamic operator [](dynamic field) => get(field);
+  dynamic operator [](Object field) => get(field);
```

Comment:
Don't think these are correct, you can have null values in Firestore document fields, e.g. they exist but values are explicitly set to `null`;

![image](https://user-images.githubusercontent.com/5347038/116988300-17f06800-acc8-11eb-8478-19011ce69fa2.png)


So maybe should be `Object?`?